tests/integration/test_install.py

```diff
@@ -136,55 +136,60 @@ def test_install_scan_ping(
     opts = InstallOptions()
     opts.no_ping = False
 
-    install_cli(opts, id_='w1')
+    await install_cli(opts, id_='w1')
     out = capsys.readouterr().out
     assert INSTALLED_MSG.format(wfrun='w1/run2') in out
     assert WF_ACTIVE_MSG.format(wf='w1') in out
     assert scan.FLOW_STATE_SYMBOLS["running"] in out
     # Empty contact file faked with "touch":
     assert f"{BAD_CONTACT_MSG} w1/run1" in caplog.text
 
-    install_cli(opts, id_='w2')
+    await install_cli(opts, id_='w2')
     out = capsys.readouterr().out
     assert INSTALLED_MSG.format(wfrun='w2/run1') in out
     assert WF_ACTIVE_MSG.format(wf='w2') not in out
 
 
-def test_install_gets_back_compat_mode_for_plugins(
-    src_run_dirs: Callable,
+async def test_install_gets_back_compat_mode_for_plugins(
+    src_run_dirs: Tuple[Path, Path],
     monkeypatch: pytest.MonkeyPatch,
+    capcall,
     capsys: pytest.CaptureFixture,
 ):
-    """Assert that pre cylc install will detect whether a workflow
+    """Assert that cylc install will detect whether a workflow
     should use back compat mode _before_ running pre_configure plugins
     so that those plugins can use that information.
     """
-    class failIfDeprecated:
+    # track calls of the check_deprecation method
+    # (this is the thing that sets cylc.flow.flags.back_compat)
+    check_deprecation_calls = capcall(
+        'cylc.flow.scripts.install.check_deprecation'
+    )
+
+    @EntryPointWrapper
+    def failIfDeprecated(*args, **kwargs):
         """A fake Cylc Plugin entry point"""
-        @staticmethod
-        def load():
-            return failIfDeprecated.raiser
-
-        @staticmethod
-        def raiser(*_, **__):
-            import cylc.flow.flags
-            if cylc.flow.flags.cylc7_back_compat:
```

Comment:
This test was intended to check that the back-compat flag was set correctly, however, it achieved this by monkeypatching the back-compat flag (i.e. manually setting it to what it expected it to be).